doc/internals/chunked-arrays.rst

```diff
@@ -0,0 +1,91 @@
+.. currentmodule:: xarray
+
+.. _internals.chunkedarrays:
+
+Alternative chunked array types
+===============================
+
+.. warning::
+
+    This is a *highly* experimental feature. Please report any bugs or other difficulties on `xarray's issue tracker <https://github.com/pydata/xarray/issues>`_.
+    In particular see discussion on `xarray issue #6807 <https://github.com/pydata/xarray/issues/6807>`_
+
+Xarray can wrap chunked dask arrays (see :ref:`dask`), but can also wrap any other chunked array type that exposes the correct interface.
+This allows us to support using other frameworks for distributed and out-of-core processing, with user code still written as xarray commands.
+
+The basic idea is that by wrapping an array that has an explicit notion of ``.chunks``, xarray can expose control over
+the choice of chunking scheme to users via methods like :py:meth:`DataArray.chunk` whilst the wrapped array actually
+implements the handling of processing all of the chunks.
+
+Chunked array requirements
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A chunked array needs to meet all the :ref:`requirements for normal duck arrays <internals.duckarrays>`, but should also implement these methods:
+
+- ``.chunk``
```

Comment:
do you mean the `.chunks` property? What about the "core operations" below? Can we add an explicit list of which operations are required for which parts of Xarray's API?

Reply:
Yes I did mean that.

So what I've done now is get RST to actually render all the attributes and methods of the `ChunkManagerEntrypoint` ABC. That way I don't have to duplicate information, I can just improve the docstrings of those methods. This should also help to avoid future inconsistencies between the docs and the implementation.

Reply:
I can't seem to get sphinx to fully render all of the docstrings, not just the one-line summary of them. It is possible because it's done in [dask.distributed's API docs page](https://distributed.dask.org/en/latest/api.html#client) ([source](https://github.com/dask/distributed/blob/main/docs/source/api.rst#client)), but I can't see how the directives I've used are any different. Is there some kind of config command that might affect this?